### DISCUSSION TITLE
Publish stream via UDP with rtsp-simple-server

### DISCUSSION THREAD

**Discussion by aduzsardi** [bluenviron/mediamtx#266] [Questions]
## Which version are you using?

v0.9.15

## Which operating system are you using?

Windows - as the publisher , running rtsp-simple-server
Linux - as the client with an ffmpeg process and a hls endpoint

## Question
There's this quiestion #82 which asks for forcing TCP connections , i would like to know the opposite , forcing UDP between the rtsp-simple-server and the ffmpeg process (in docker) running on the linux host

Do you have an example configuration for this ? Like what ports are required to be opened between the hosts , what configuration options should be enabled or disabled and maybe ffmpeg flags that need to be added.

## Description
I'm asking this because the connection between the linux host and the windows host is not the most reliable and are in two remote locations from each-other , so tcp will most likely not be the best option to stream video across the VPN

**Comment by aler9** [comment]
Hello,
If the system architecture is the one you've written, i think you want to "read streams with UDP", rather than publishing streams with UDP. To use UDP when reading with ffmpeg, use the `-rtsp-transport udp` flag before the input:
```
ffmpeg -rtsp_transport udp -i rtsp://server/path ...
```

To force all clients to use UDP, disable TCP in the configuration of `rtsp-simple-server`:
```yml
protocols: [tcp]
```

Regarding UDP inside a VPN, it will perform in the way you want (an unreliable connection which allows frame drops) only if the VPN is based on UDP too - if the VPN is TCP-based, UDP packets will be encapsulated in TCP packets, which do not allow frame drops, so forcing UDP is useless.
I can suggest [wireguard](https://www.wireguard.com/), which is a UDP-based VPN with clients for both Windows and Linux.

**Comment by aduzsardi** [comment]
Unfortunately i can't decide what protocol and/or software is used for the VPN setup.

Thank you for the clarification, i'll have to ask what protocol is used on the VPN

[a comment by an automated account is not shown]
